server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/GlobalOrdinalsStringTermsAggregator.java

```diff
@@ -420,4 +392,141 @@ public boolean advanceExact(int target) throws IOException {
             return false;
         }
     }
+
+    /**
+     * Strategy for collecting global ordinals.
```

Comment:
I think your comment from the PR description that we have two collection strategies and each has two iteration strategies would fit well in the javadoc for this class.  Or at the very least, a note on `forEach` that it should account for both iteration cases.

Reply:
:+1: